src/store/Harvesting.ts

```diff
@@ -328,40 +353,127 @@ export default {
+        ) {
+            const harvestingModel = harvestingService.getHarvestingModel(accountAddress);
+            harvestingService.updateNewAccountLinkKeyInfo(harvestingModel, {
+                newEncRemotePrivateKey,
+                newRemotePublicKey,
+            });
+            commit('currentSignerHarvestingModel', harvestingModel);
+        },
         UPDATE_VRF_ACCOUNT_PRIVATE_KEY(
             { commit },
             { accountAddress, encVrfPrivateKey }: { accountAddress: string; encVrfPrivateKey: string },
         ) {
-            const harvestingService = new HarvestingService();
             const harvestingModel = harvestingService.getHarvestingModel(accountAddress);
             harvestingService.updateVrfKey(harvestingModel, encVrfPrivateKey);
             commit('currentSignerHarvestingModel', harvestingModel);
         },
+        UPDATE_NEW_VRF_KEY_INFO(
+            { commit },
+            {
+                accountAddress,
+                newEncVrfPrivateKey,
+                newVrfPublicKey,
+            }: { accountAddress: string; newEncVrfPrivateKey: string; newVrfPublicKey: string },
+        ) {
+            const harvestingModel = harvestingService.getHarvestingModel(accountAddress);
+            harvestingService.updateNewAccountLinkKeyInfo(harvestingModel, { newEncVrfPrivateKey, newVrfPublicKey });
+            commit('currentSignerHarvestingModel', harvestingModel);
+        },
         UPDATE_HARVESTING_REQUEST_STATUS(
             { commit },
             { accountAddress, delegatedHarvestingRequestFailed }: { accountAddress: string; delegatedHarvestingRequestFailed: boolean },
         ) {
-            const harvestingService = new HarvestingService();
             const harvestingModel = harvestingService.getHarvestingModel(accountAddress);
             harvestingService.updateDelegatedHarvestingRequestFailed(harvestingModel, delegatedHarvestingRequestFailed);
             commit('currentSignerHarvestingModel', harvestingModel);
         },
+        UPDATE_LOCAL_ACCOUNT_LINK_PRIVATE_KEY(
+            { dispatch },
+            {
+                currentSignerAccountInfo,
+                currentSignerHarvestingModel,
+            }: { currentSignerAccountInfo: AccountInfo; currentSignerHarvestingModel: HarvestingModel },
+        ) {
+            const { address, supplementalPublicKeys } = currentSignerAccountInfo;
+            const accountAddress = address.plain();
+            const { linked, vrf } = supplementalPublicKeys;
+            const {
+                newEncRemotePrivateKey,
+                newEncVrfPrivateKey,
+                newRemotePublicKey,
+                newVrfPublicKey,
+                encRemotePrivateKey,
+                encVrfPrivateKey,
+            } = currentSignerHarvestingModel;
+
+            // If remote key link not exist from network or local
+            // Set local encRemotePrivateKey to null
+            if (!linked || !newRemotePublicKey) {
```

Comment:
should be `&&` instead of `||`, no?

Reply:
I believe use `||` is more suitable in this situation.
because if 1 of them is empty, just set to `null`